### ISSUE TITLE
Recreated instance does appears as old instance

### ISSUE THREAD

**Issue by TheGeka** [mastodon/mastodon#32565]
### Steps to reproduce the problem

1. Create instance with subdomain url (subdomain.domain.tld
2. Nuke the instance and create a new one without the subdomain (domain.tld)

### Expected behaviour

Recreated instance behaves normal

### Actual behaviour

Recreated instance appears like the old one

### Detailed description

Uhm i have an interesting problem. Originally i used the LOCAL_DOMAIN mastodon.bloodmoon-network.de but since i wanted only bloodmoon-network.de i decided to delete the old instance and create a new one. 
The issue now is that on my instance everything looks fine and i have the correct handle (@Laelia@Bloodmoon-Network.de) but on i.e. mastodon.social it still says the old handle (@Laelia@mastodon.bloodmoon-network.de).

Local:
![Image](https://github.com/user-attachments/assets/2e1f97e1-b939-4d9f-8cc8-3d3312a164dc)
On Mastodon.Social
![Image](https://github.com/user-attachments/assets/3474bcaf-ece8-46d2-af46-bec5e4b4be98)

Any idea how to fix that

### Mastodon instance

bloodmoon-network.de

### Mastodon version

v4.3.0

### Environment

Deployed using the docker images on kubernetes
```
LOCAL_DOMAIN=bloodmoon-network.de
WEB_DOMAIN=mastodon.bloodmoon-network.de
```
Nginx redirect for well know because LOCAL_DOMAIN and WEB_DOMAIN differ
```
        location = /.well-known/host-meta {
            return 301 https://mastodon.bloodmoon-network.de$request_uri;
        }
        location /.well-known/webfinger {
            add_header Access-Control-Allow-Origin '*';
            return 301 https://mastodon.bloodmoon-network.de$request_uri;
        }
        ## mastodon web url
        location ~ ^/(about/more|settings|web|pghero|sidekiq|admin|interact|explore|public|@.*|relationships|filters|terms|inert\.css)$ {
            return 301 https://mastodon.bloodmoon-network.de$request_uri;
        }
        ## mastodon .well-known
        location ~ ^/(.well-known/(host-meta|nodeinfo|webfinger|change-password|keybase-proof-config)|nodeinfo) {
            return 301 https://mastodon.bloodmoon-network.de$1 permanent;
        }
```

**Comment by renchap** [comment]
How did you destroy your old instance? Did you use the self-destruct process, and let it go to completion?

**Comment by TheGeka** [comment]
I just found another issues talking about domain change #5774 where someone said the self-destruct command is what you need. Originally i just deleted the data without the self-destruct command. Rn the self destruct is running on the current domain and i guess need to run it again on the old domain once it's done.

**Comment by renchap** [comment]
Yes, and you need to run it against the database from the first domain.

Basically the way ActivityPub works is that your domain name has been stored in a lot of other instances, along with its public key. The self-destruct process sends a (signed) notification to every known instance that you are self-destructing, and they should remove everything about you. If this is not done (or not processed), then the other instances will already know your domain with your *old* public key, and reject anything signed by your *new* public key.

If you do not have a database backup of your old instance containing your key, then unfortunately you will not be able to use your domain for a new instance, or you will need to somehow ask all the other instance admins to manually remove your domain from their instance.

**Comment by TheGeka** [comment]
Hm that is sad cause the original database was just deleted as i though i can do it the same way as i did with my matrix server where i also recreated it to change it from a subdomain to the base domain. 

Does that really mean the whole domain is now unusable or would i be able to create a new instance under a different subdomain?

**Comment by TheGeka** [comment]
Update: I managed to unstuck my domain by doing multiple passes of  self destruct -> recreate -> self destruct using both the sub.domain.tld and domain.tld. Now everything seems to work except Threads federation(Maybe because the instance is brand new?) and the bluesky bridge which i know filter new accounts
